File: lint-arwaky/archive/surfaces/mcp_server_handler.py
"""Main entry point for the Auto Linter MCP server.

This module exposes a FastMCP server that registers all tool
operations defined in the AES architecture.
"""

import os
from dotenv import load_dotenv
from ..contract import ServiceContainerAggregate
from mcp.server.fastmcp import FastMCP
from .mcp_tools_store import register_tools

load_dotenv()


class McpServerHandlerSurface:
    """Main surface for the Auto Linter MCP server."""

    def __init__(self) -> None:
        self.server_name = "auto-linter"
        self.mcp = FastMCP(self.server_name)

    def _setup_environment(self) -> None:
        """Ensure virtual environment bin is in the system PATH."""
        import sys

        venv_bin = os.path.dirname(sys.executable)
        current_path = os.environ.get("PATH", "")
        if venv_bin not in current_path:
            os.environ["PATH"] = venv_bin + os.path.pathsep + current_path

    def run_server(self, container: ServiceContainerAggregate) -> None:
        """Run the MCP server fulfillment."""
        self._setup_environment()
        register_tools(self.mcp, container)
        self.mcp.run()

    def run(self, container: ServiceContainerAggregate) -> None:
        """Alias for run_server for compatibility."""
        self.run_server(container)

File: lint-arwaky/src-rust/surfaces/mcp_server_handler.rs
/// Main entry point for the Auto Linter MCP server.
use crate::taxonomy::*;
use crate::contract::*;
use crate::surfaces::mcp_tools_store::register_tools;

pub struct McpServerHandlerSurface;

impl McpServerHandlerSurface {
    pub fn new() -> Self {
        Self
    }

    pub fn run_server(&self, container: ServiceContainerAggregate) {
        // In real impl: create FastMCP equivalent, register tools, run
        register_tools(container);
        println!("Auto-Linter MCP server starting...");
        println!("Server name: auto-linter");
        println!("Note: Full MCP server requires 'fastmcp' / 'mcp' crate integration");
    }

    pub fn run(&self, container: ServiceContainerAggregate) {
        self.run_server(container);
    }
}
